gateway/reverse_proxy.go

```diff
@@ -923,6 +923,17 @@ func (p *ReverseProxy) handleOutboundRequest(roundTripper *TykRoundTripper, outr
 		latency = time.Since(begin)
 	}()
 
+	if p.TykAPISpec.HasMock {
```

Comment:
Same as above, imho should be standard middleware.

Reply:
HasMock does loops and checks every paths whether there is a mock. Why would it check for a normal request by looping everytime?